Lib/inspect.py

```diff
@@ -265,7 +265,7 @@ def isgetsetdescriptor(object):
         return False
 
 def isfunction(object):
```

Comment:
This change can cause lots of problems to existing users. Because `inspect` is already used in quite complex code, which uses introspection of different objects in a very specific way.

Reply:
There is no functional change in `isfunction`. I just thought it's good to clarify that only true Python functions are matched.

Reply:
What about functions defined in PyPy, RustPython, and others? What about PyO3 functions? What about other possible binary ways of defining functions? 

> There is no functional change in `isfunction`

This would need a proof with real data, unfortunatelly :(

Reply:
> There is no functional change in isfunction
> This would need a proof with real data, unfortunatelly :(

The github review interface might be confusing here: the only change in `isfunction` is in the docstring. The "real" change is in `getfile` and `findsource`.